subgraphs/balancer-v2/src/common/constants.ts

```diff
@@ -0,0 +1,172 @@
+  export const AVALANCHE = "AVALANCHE";
+  export const AURORA = "AURORA";
+  export const BSC = "BINANCE_SMART_CHAIN";
+  export const CELO = "CELO";
+  export const CRONOS = "CRONOS";
+  export const ETHEREUM = "ETHEREUM";
+  export const FANTOM = "FANTOM";
+  export const HARMONY = "HARMONY_SHARD_0";
+  export const MOONBEAM = "MOONBEAM";
+  export const MOONRIVER = "MOONRIVER";
+  export const OPTIMISM = "OPTIMISTIC_ETHEREUM";
+  export const POLYGON = "POLYGON_POS";
+  export const XDAI = "XDAI";
+}
+
+export namespace ProtocolType {
+  export const EXCHANGE = "EXCHANGE";
+  export const LENDING = "LENDING";
+  export const YIELD = "YIELD";
+  export const BRIDGE = "BRIDGE";
+  export const GENERIC = "GENERIC";
+}
+
+export namespace VaultFeeType {
+  export const MANAGEMENT_FEE = "MANAGEMENT_FEE";
+  export const PERFORMANCE_FEE = "PERFORMANCE_FEE";
+  export const DEPOSIT_FEE = "DEPOSIT_FEE";
+  export const WITHDRAWAL_FEE = "WITHDRAWAL_FEE";
+}
+
+export namespace LiquidityPoolFeeType {
+  export const FIXED_TRADING_FEE = "FIXED_TRADING_FEE";
+  export const TIERED_TRADING_FEE = "TIERED_TRADING_FEE";
+  export const DYNAMIC_TRADING_FEE = "DYNAMIC_TRADING_FEE";
+  export const FIXED_PROTOCOL_FEE = "FIXED_PROTOCOL_FEE";
+  export const DYNAMIC_PROTOCOL_FEE = "DYNAMIC_PROTOCOL_FEE";
+}
+
+export namespace RewardTokenType {
+  export const DEPOSIT = "DEPOSIT";
+  export const BORROW = "BORROW";
+}
+//////////////////////////////
+///// Ethereum Addresses /////
+//////////////////////////////
+
+export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const ETH_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
+
+////////////////////////
+///// Type Helpers /////
+////////////////////////
+
+export const DEFAULT_DECIMALS = 18;
+
+export const BIGINT_ZERO = BigInt.fromI32(0);
+export const BIGINT_ONE = BigInt.fromI32(1);
+export const BIGINT_TWO = BigInt.fromI32(2);
+export const BIGINT_THOUSAND = BigInt.fromI32(1000);
+export const BIGINT_MAX = BigInt.fromString(
+  "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+);
+
+export const INT_ZERO = 0 as i32;
+export const INT_ONE = 1 as i32;
+export const INT_TWO = 2 as i32;
+
+export const BIGDECIMAL_ZERO = new BigDecimal(BIGINT_ZERO);
+export const BIGDECIMAL_ONE = new BigDecimal(BIGINT_ONE);
+export const BIGDECIMAL_TWO = new BigDecimal(BIGINT_TWO);
+
+export const MAX_UINT = BigInt.fromI32(2).times(BigInt.fromI32(255));
+
+/////////////////////
+///// Date/Time /////
+/////////////////////
+
+export const SECONDS_PER_DAY = 60 * 60 * 24; // 86400
+export const MS_PER_DAY = new BigDecimal(BigInt.fromI32(24 * 60 * 60 * 1000));
+export const DAYS_PER_YEAR = new BigDecimal(BigInt.fromI32(365));
+export const MS_PER_YEAR = DAYS_PER_YEAR.times(new BigDecimal(BigInt.fromI32(24 * 60 * 60 * 1000)));
+
+/////////////////////////////
+///// Protocol Specific /////
+/////////////////////////////
+
+export class AddressByNetwork {
```

Comment:
Why not group by network instead? e.g.
```
class AddressByNetwork {
  vaultAddress: string;
  wethAddress: string;
  ...
}
```
and instantiate one per network?

This seems more scalable as we have more network-specific fields.

Reply:
Also, for network specific implementation, you should do something similar to Yearn: https://github.com/yearn/yearn-vaults-v2-subgraph

Reply:
yep, both are correct